### ISSUE TITLE
Define slots explicitly

### ISSUE THREAD

**Issue by bdestombe** [dtscalibration/python-dts-calibration#132]
**Describe the bug**
Upon importing the library:
```python
/srv/conda/envs/notebook/lib/python3.7/typing.py:845: FutureWarning: xarray subclass DataStore should explicitly define __slots__
  super().__init_subclass__(*args, **kwargs)
```

**To Reproduce**
Import the dtscalibration library

**Expected behavior**
No warnings.

**Additional context**
Xarray, does not recommend subclassing (and thus using slots), but instead to put everything under an accessor:
```python
ds.dts.single_ended_calibration()
```
For reasons listed here: `http://xarray.pydata.org/en/stable/internals.html#extending-xarray`
Mostly needed when combining multiple 'extensions' to xarray. But to my opinion, this requires a lot of work, breaks the current version, and would not change the functionality of the package.

**Comment by BSchilperoort** [comment]
As for now I have decided to specifically silence this warning, as it just clutters the output and could confuse users. We might have to keep watching what xarray decides to do, but they seem to not have made any changes or decisions over the last year.

**Comment by BSchilperoort** [comment]
@bdestombe do we still need to tackle this issue? If our way of subclassing xarray is deprecated, it could break upon any new release of xarray. This might be a thing we should do before v3.0 :pensive:

**Comment by bdestombe** [comment]
Hi Bart, this has been the case for several years now. Furthermore, I don't see any urgent deprecation warnings and therefor also don't think we have to tackle this now.

Instead, let's build an accessor in parallel to the current implementation.

**Comment by BSchilperoort** [comment]
I have been trying out accessors, and one problem is that it's difficult for code analyzers to determine what the type and available methods are of xarray/pandas extensions: https://github.com/microsoft/pylance-release/issues/1112

In VS Code, the code analyzer does not see that `regrid` is a class under `ds` (Dataset), and that `to_dataset` is a valid function. No docstrings are displayed either:
![image](https://github.com/dtscalibration/python-dts-calibration/assets/12114825/5f5b0f0f-e2f0-468c-8cb5-c9ff18b50611)

**Comment by BSchilperoort** [comment]
Fixed by #204
